code-challenges.test.js

```diff
@@ -20,33 +20,361 @@ const secretCodeWord2 = "Gobbledygook"
+
+//   expect(received).toEqual(expected) // deep equality
+
+//   Expected: "L4ck4d41s1c4l"
+//   Received: undefined
+
+//     38 |         const secretCodeWord5 = "Supercalifragilisticexpialidocious"
+//     39 |
+//   > 40 |         expect(superSecretCodingMachine(secretCodeWord1)).toEqual("L4ck4d41s1c4l"),
+//        |                                                           ^
+//     41 |         expect(superSecretCodingMachine(secretCodeWord2)).toEqual("G0bbl3dyg00k"),
+//     42 |         expect(superSecretCodingMachine(secretCodeWord3)).toEqual("3cc3ntr1c"),
+//     43 |         expect(superSecretCodingMachine(secretCodeWord4)).toEqual("3333"),
+
+//     at Object.toEqual (code-challenges.test.js:40:59)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+
+// jest-test-3:
+// PASS  ./code-challenges.test.js
+// superSecretCodingMachine
+//   ✓ take in a string and returns a string with a coded message (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
 
 // b) Create the function that makes the test pass.
 
+// Comment: The below .replaceAll's is where my head initially went. I liked the regex and how overall the code was relatively short but I didn't like how I had to repeat .replaceAll multiple times, though I had to due to the constraints of regex only being only able to pass one replacement value per expression. With some additional research, I was able to get the following two functions using an object to work ALMOST correctly. They work however I ran into an issue where even though I was using the /i to catch both upper and lowercase letterA, the "E" in the secretCodeWord3 was returning undefined. I am still trying to figure out why but I was able to add an additional key:value pair to the object as a work around so that it returns as required. I chose this route because I thought it was a cleaner looking fix as opposed to adding a .toUpperCase or .toLowerCase. I am going to figure out the issue this week as I have extra time.
+
+// const superSecretCodingMachine = (string) => {
+//     return string.replaceAll(/a/gi, 4).replaceAll(/e/gi, 3).replaceAll(/i/gi, 1).replaceAll(/o/gi, 0)
+// }
+
+// const superSecretCodingMachine = (string) => {
+//     regex = /a|e|E|i|o|u/gi;
+//     const superSecretCode = {
+//         "a": 4,
+//         "e": 3,
+//         "E": 3,
+//         "i": 1,
+//         "o": 0,
+//       };
+//     return string.replaceAll(regex, (matched) => superSecretCode[matched])
+// }
+
+const superSecretCodingMachine = (string) => {
+    const superSecretCode = {
+        "a": 4,
+        "e": 3,
+        "E": 3,
+        "i": 1,
+        "o": 0,
+        "u": 6,
+      };
+    return string.replaceAll(/a|e|E|i|o|u/gi, (matched) => superSecretCode[matched])
+}
+
+// This one below was done AFTER initial assessment submission. It has a couple additional things cleaned up. The one above passes but I was not happy with it. 
+
```

Comment:
I love the refactor AND the documentation, GREAT work Josh. I completely agree with you